sys/net/network_layer/ng_sixlowpan/ctx/ng_sixlowpan_ctx.c

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2015 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ */
+
+#include <stdbool.h>
+#include <inttypes.h>
+
+#include "mutex.h"
+#include "net/ng_sixlowpan/ctx.h"
+#include "vtimer.h"
+
+static ng_sixlowpan_ctx_t _ctxs[NG_SIXLOWPAN_CTX_SIZE];
+static uint32_t _ctx_inval_times[NG_SIXLOWPAN_CTX_SIZE];
+static mutex_t _ctx_mutex = MUTEX_INIT;
+
+static uint32_t _current_minute(void);
+static void _update_lifetime(unsigned int id);
+
+static inline bool _still_valid(unsigned int id)
+{
+    _update_lifetime(id);
+    return (_ctxs[id].ltime > 0);
+}
+
+ng_sixlowpan_ctx_t *ng_sixlowpan_ctx_lookup_addr(const ng_ipv6_addr_t *addr)
+{
+    mutex_lock(&_ctx_mutex);
+
+    for (unsigned int i = 0; i < NG_SIXLOWPAN_CTX_SIZE; i++) {
+        if ((_ctxs[i].ltime > 0) &&
+            _ctxs[i].prefix_len <= ng_ipv6_addr_match_prefix(&_ctxs[i].prefix, addr)) {
+            if (_still_valid(i)) {
+                mutex_unlock(&_ctx_mutex);
+                return _ctxs + i;
```

Comment:
(my point was: a stupid compiler would first dereference and then reference the pointer again).